pkgs/development/python-modules/pytest-ansible/default.nix

```diff
@@ -4,31 +4,51 @@
 , ansible-core
 , coreutils
 , coverage
+, pytest
 , pytestCheckHook
+, pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "pytest-ansible";
   version = "3.0.0";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "ansible";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-kxOp7ScpIIzEbM4VQa+3ByHzkPS8pzdYq82rggF9Fpk=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-kxOp7ScpIIzEbM4VQa+3ByHzkPS8pzdYq82rggF9Fpk=";
   };
 
   postPatch = ''
-    substituteInPlace tests/conftest.py inventory --replace '/usr/bin/env' '${coreutils}/bin/env'
+    substituteInPlace tests/conftest.py inventory \
+      --replace '/usr/bin/env' '${coreutils}/bin/env'
   '';
 
-  nativeCheckInputs =  [ pytestCheckHook coverage ];
-  propagatedBuildInputs = [ ansible-core ];
+  buildInputs = [
+    pytest
```

Comment:
How I could know it was missing ? pythonImportsCheck Did help ? Does it also check for propagatedBuildInputs the docs wasn't clear

Reply:
For us, it would works without `pytest` as we use pytest extensions/plugins/addons usually with `pytestCheckHook` for nikpgs.

`pytest` is a runtime requirement (should be part of `propagatedBuildInputs`) but to be able to override `pytest` it has to go to `buildInputs`.